File: MigrationPy/pipelines/p2/code/p1/graph/ds1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from p1.config.ConfigStore import *
from p1.udfs.UDFs import *

def ds1(spark: SparkSession) -> DataFrame:
    return spark.read\
        .schema(StructType([StructField("id1", StringType(), True), StructField("id2", StringType(), True)]))\
        .option("header", True)\
        .option("sep", ",")\
        .csv("dede")

File: scalaGemMigrationExamples/pipelines/p1/code/src/main/scala/io/prophecy/pipelines/p1/graph/ds1.scala
package io.prophecy.pipelines.p1.graph

import io.prophecy.libs._
import io.prophecy.pipelines.p1.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object ds1 {

  def apply(context: Context): DataFrame =
    context.spark.read
      .format("csv")
      .option("header",                          true)
      .option("sep",                             ",")
      .schema(StructType(Array(StructField("id", StringType, true))))
      .load("de")

}
